app/code/Magento/Catalog/Controller/Adminhtml/Product/Set/Edit.php

```diff
@@ -6,53 +6,77 @@
  */
 namespace Magento\Catalog\Controller\Adminhtml\Product\Set;
 
-use Magento\Framework\App\Action\HttpGetActionInterface as HttpGetActionInterface;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Registry;
+use Magento\Backend\App\Action\Context;
+use Magento\Backend\Model\View\Result\Page;
+use Magento\Framework\View\Result\PageFactory;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Eav\Model\Entity\Attribute\SetFactory;
+use Magento\Catalog\Controller\Adminhtml\Product\Set;
+use Magento\Framework\App\Action\HttpGetActionInterface;
 
-class Edit extends \Magento\Catalog\Controller\Adminhtml\Product\Set implements HttpGetActionInterface
+/**
+ * Class Edit
+ *
+ * @package Magento\Catalog\Controller\Adminhtml\Product\Set
+ */
+class Edit extends Set implements HttpGetActionInterface
 {
     /**
-     * @var \Magento\Framework\View\Result\PageFactory
+     * @var PageFactory
      */
     protected $resultPageFactory;
 
     /**
-     * @param \Magento\Backend\App\Action\Context $context
-     * @param \Magento\Framework\Registry $coreRegistry
-     * @param \Magento\Framework\View\Result\PageFactory $resultPageFactory
+     * @var SetFactory
+     */
+    protected $setFactory;
+
+    /**
+     * Edit constructor.
+     *
+     * @param Context $context
+     * @param Registry $coreRegistry
+     * @param PageFactory $resultPageFactory
+     * @param SetFactory $setFactory
      */
     public function __construct(
-        \Magento\Backend\App\Action\Context $context,
-        \Magento\Framework\Registry $coreRegistry,
-        \Magento\Framework\View\Result\PageFactory $resultPageFactory
+        Context $context,
+        Registry $coreRegistry,
+        PageFactory $resultPageFactory,
+        SetFactory $setFactory = null
     ) {
         parent::__construct($context, $coreRegistry);
         $this->resultPageFactory = $resultPageFactory;
+        $this->setFactory = $setFactory ?: ObjectManager::getInstance()->get(SetFactory::class);
     }
 
     /**
-     * @return \Magento\Backend\Model\View\Result\Page
+     * Controller execute method
+     *
+     * @return ResultInterface
      */
     public function execute()
     {
         $this->_setTypeId();
-        $attributeSet = $this->_objectManager->create(\Magento\Eav\Model\Entity\Attribute\Set::class)
-            ->load($this->getRequest()->getParam('id'));
-
+        $attributeSet = $this->setFactory->create()->load($this->getRequest()->getParam('id'));
         if (!$attributeSet->getId()) {
-            return $this->resultRedirectFactory->create()->setPath('catalog/*/index');
+            return $this->resultFactory->create(ResultFactory::TYPE_REDIRECT)->setPath('catalog/*/index');
```

Comment:
What is the benefit in not using two separate factories?

Reply:
Using a single common already-inherited `$resultFactory` gives more consistency and a single unified approach toward creating Result objects. Isn't this usually the preferred method since all controllers have an inherited `$resultFactory`? Also, this gives scope in the future to phase out unnecessary independent Result factories in the constructor dependencies, since many controllers are already doing this.

Reply:
@drpayyne they both are already inherited.

Please revert or propose not using `$this->resultRedirectFactory` in https://github.com/magento/architecture/issues and enforce with a static test.